torchvision/prototype/features/_image.py

```diff
@@ -32,6 +32,31 @@ def from_pil_mode(cls, mode: str) -> ColorSpace:
         else:
             return cls.OTHER
 
+    @staticmethod
+    def from_tensor_shape(shape: List[int]) -> ColorSpace:
```

Comment:
I think this makes more sense to be on the Enum than the method. At any case it was moved because JIT didn't like it to be a class method. We don't have to keep it in the enum, we just need to have it as a standalone method.

Reply:
So we can't even have classmethods on objects that are not covered by JIT? :exploding_head:

Reply:
Plus, I intentionally put the term "guess" in the name, since the number of channels is not sufficient to pick the right colorspace. For example, CMYK also has 4 channels, but would be classified as RGBA. However, this is not a problem now since we don't support it yet and maybe never will.

Reply:
If it works as is, I wouldn't mess with this class any further. If it is needed we can remove the `: Any` though.

Reply:
@pmeier I'm also really flexible. I don't have context why the `tensor_contents` is introduced and what's supposed to be. Would you like to send a PR once this is merged to see if you can make it work in the original location?